src/test/java/eu/dissco/orchestration/backend/service/SourceSystemServiceTest.java

```diff
@@ -339,13 +342,22 @@ void testUpdateSourceSystem() throws Exception {
     var updateCron = mock(APIreplaceNamespacedCronJobRequest.class);
     given(batchV1Api.replaceNamespacedCronJob(anyString(), eq(NAMESPACE), any(V1CronJob.class)))
         .willReturn(updateCron);
+    if (triggerTranslator) {
+      var createJob = mock(APIcreateNamespacedJobRequest.class);
+      given(
+          batchV1Api.createNamespacedJob(eq(NAMESPACE), any(V1Job.class))).willReturn(createJob);
+    }
 
     // When
-    var result = service.updateSourceSystem(BARE_HANDLE, sourceSystem, OBJECT_CREATOR, SYSTEM_PATH);
+    var result = service.updateSourceSystem(BARE_HANDLE, sourceSystem, OBJECT_CREATOR, SYSTEM_PATH,
+        triggerTranslator);
 
     // Then
     assertThat(result).isEqualTo(expected);
     then(repository).should().updateSourceSystem(givenSourceSystem(2));
+    if (!triggerTranslator) {
+      then(batchV1Api).shouldHaveNoMoreInteractions();
+    }
```

Comment:
i think you need to have the opposite condition here too for completeness

```
else {
      then(batchV1Api).should().createNamespacedCronJob(any(), any());
    }
```

Reply:
Added although the given also does an argument check so I think it is a bit double but anyway can't hurt.